### PR TITLE
add vim syntax support

### PR DESCRIPTION
**Pull request by xdch47**
- added vim syntax support  as vimplugin in a first shot
- syntastic integration with nagel can follow

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -125,6 +127,7 @@ Fine tuning of the installation directories:
   --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
   --mandir=DIR            man documentation [DATAROOTDIR/man]
   --docdir=DIR            documentation root [DATAROOTDIR/doc]
+  --vimplugindir=DIR      vim plugin directory [DATAROOTDIR/share/vim/vimfile]
```
[the remark this answers is not in the document]
Comment (xdch47) [reply configure]: Fixed - sorry, didn't run `./configure --help`
Thanks !

File: contrib/vim/ftdetect/modules.vim [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+au BufNewFile,BufRead *
+    \ if (getline(1) =~? "^#%Module") |
+    \     set filetype=modules |
```
[the remark this answers is not in the document]
Comment (xdch47) [reply contrib/vim/ftdetect/modules.vim]: You are right `modulefile` is a way better as name

File: contrib/vim/ftplugin/modules.vim [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+endif
+let b:did_modules_ftplugin = 1
+
+set syntax=modules
+
+" from tcl ftplugin:
+" Vim filetype plugin file
+" Language:         Tcl
+" Maintainer:       Robert L Hicks <sigzero@gmail.com>
+" Latest Revision:  2009-05-01
```
[the remark this answers is not in the document]
Comment (xdch47) [reply contrib/vim/ftplugin/modules.vim]: removed

File: Makefile [hunk 4]
```diff
@@ -240,6 +240,10 @@ ifeq ($(compatversion),y)
 	cp $(COMPAT_DIR)/ChangeLog $(DESTDIR)$(docdir)/ChangeLog-compat
 	cp $(COMPAT_DIR)/NEWS $(DESTDIR)$(docdir)/NEWS-compat
 endif
+endif
+ifeq ($(vimplugin),y)
+	mkdir -p $(DESTDIR)$(vimplugindir)
+	cp -r contrib/vim/* $(DESTDIR)$(vimplugindir)/
```
[the remark this answers is not in the document]
Comment (xdch47) [reply Makefile]: used `install -D` command (coreutils)

**Comment by xdch47** [comment]
Thanks for your review! Hopefully it's better now.
Do you use the `vim`?  We could add a few lines for integrating `modulefile` (`tcl`) linting via [vim-syntastic](https://github.com/vim-syntastic/syntastic) and [nagelfar](http://nagelfar.sourceforge.net/) but maybe thats overkill, since a module file should not be a complex script at all.

Btw. I didn't forgot the `zsh`-completion issues...

File: configure [hunk 5]
```diff
@@ -125,6 +127,7 @@ Fine tuning of the installation directories:
   --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
   --mandir=DIR            man documentation [DATAROOTDIR/man]
   --docdir=DIR            documentation root [DATAROOTDIR/doc]
+  --vimplugindir=DIR      vim plugin directory [DATAROOTDIR/vim/vimfile]
```
Comment (xdelaruelle) [inline configure]: `s` is missing in `[DATAROOTDIR/vim/vimfile]` (vimfiles)

File: Makefile [hunk 6]
```diff
@@ -240,6 +240,12 @@ ifeq ($(compatversion),y)
 	cp $(COMPAT_DIR)/ChangeLog $(DESTDIR)$(docdir)/ChangeLog-compat
 	cp $(COMPAT_DIR)/NEWS $(DESTDIR)$(docdir)/NEWS-compat
 endif
+endif
+ifeq ($(vimplugin),y)
+	mkdir -p $(DESTDIR)$(vimplugindir)
+	install -D contrib/vim/ftdetect/modulefile.vim  $(DESTDIR)$(vimplugindir)/ftdetect/modulefile.vim
+	install -D contrib/vim/ftplugin//modulefile.vim $(DESTDIR)$(vimplugindir)/ftplugin/modulefile.vim
+	install -D contrib/vim/syntax/modulefile.vim    $(DESTDIR)$(vimplugindir)/syntax/modulefile.vim
```
Comment (xdelaruelle) [inline Makefile]: I would prefer not relying on an additional tool to install and having these files and dirs installed the same way other files here does (mkdir+cp)

File: Makefile [hunk 7]
```diff
@@ -263,6 +269,13 @@ endif
 	rm -f $(DESTDIR)$(bindir)/add.modules
 	rm -f $(DESTDIR)$(bindir)/modulecmd
 	rm -f $(DESTDIR)$(bindir)/mkroot
+ifeq ($(vimplugin),y)
+	rm -f $(DESTDIR)$(vimplugindir)/ftdetect/modulefile.vim
+	rm -f $(DESTDIR)$(vimplugindir)/ftplugin/modulefile.vim
+	rm -f $(DESTDIR)$(vimplugindir)/syntax/modulefile.vim
+	-rmdir $(DESTDIR)$(vimplugindir)/{ftdetect,ftplugin,syntax}
```
Comment (xdelaruelle) [inline Makefile]: the braces `{}` are not understood by make (you can see [the make uninstall failing on the CI](https://travis-ci.org/cea-hpc/modules/builds/466635254?utm_source=github_status&utm_medium=notification))

Comment (xdch47) [reply Makefile]: Mmh, interesting - I guess it's more a shell thing … I tested with `zsh` / `tcsh`  and both supported it, so it worked fine… maybe `sh` does not…

Comment (xdch47) [reply Makefile]: looks like the other problems… anyway replaced it

**Comment by xdch47** [comment]
Okay - Thank you !

**Pull request merged**